### PR TITLE
feat: list of submissions UI

### PR DESCRIPTION
**Pull request by andrii-uhryn-absa**
Screenshot 2023-11-15 at 10 53 40
Screenshot 2023-11-15 at 10 53 25

### CODE REVIEW

File: packages/frontend/src/app/page.tsx [hunk 1]
```diff
@@ -10,7 +10,12 @@ import { Table, TableBody, TableHeader } from '@/common/components/Table'
 
 async function getEntities(): Promise<Entity[]> {
   try {
-    const response: Response = await fetch('http://localhost:3000/api/registry')
+    const response: Response = await fetch(
```
Comment (jakubkoci) [inline packages/frontend/src/app/page.tsx:13]: Do we have to annotate it with`Response`?

File: packages/frontend/src/app/page.tsx [hunk 2]
```diff
@@ -10,7 +10,12 @@ import { Table, TableBody, TableHeader } from '@/common/components/Table'
...
+      'http://localhost:3000/api/registry',
+      {
+        cache: 'no-cache',
+      },
+    )
     const responseJson: TrustRegistry = await response.json()
```
Comment (jakubkoci) [inline packages/frontend/src/app/page.tsx:19]: Do we have to annotate it with `TrustRegistry`?

Ideally, we should rely on type inference as much as possible.

File: packages/frontend/src/app/submissions/page.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+                </td>
+                <td className="p-0">
+                  <div className="p-4 bg-white mb-1 ">
+                    <p className="leading-6 text-right min-h-6">
+                      {dayjs(item.updatedAt).format('DD/MM/YYYY')}
+                    </p>
+                  </div>
+                </td>
+                <td className="p-0">
+                  <div className="p-4 bg-white mb-1 rounded-r-lg">
```
Comment (jakubkoci) [inline packages/frontend/src/app/submissions/page.tsx:73]: Shouldn't be `rounded-r-lg` only set for the last item in the table?

File: packages/frontend/src/common/components/Filter.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
+import React, { ReactNode } from 'react'
+import { TextSmBold } from '@/common/components/Typography'
+import { PlusIcon } from '@/app/submissions/images/PlusIcon'
```
Comment (jakubkoci) [inline packages/frontend/src/common/components/Filter.tsx:3]: We should not import from `submissions` into `common` module

File: packages/frontend/src/app/submissions/page.tsx [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+import { PageContainer } from '@/common/components/PageContainer'
+import { Table, TableBody, TableHeader } from '@/common/components/Table'
+import { Filter, FilterButton } from '@/common/components/Filter'
+import { betterFetch } from '@/api'
+
+export default function Page() {
+  const [submissions, setSubmissions] = useState<Submission[]>([])
+  const [filterValues, setFilterValues] = useState<string[]>([])
+
+  function toggleFilterValue(filter: string): void {
```
Comment (jakubkoci) [inline packages/frontend/src/app/submissions/page.tsx:17]: What if we move the state of the filter and this logic inside the `Filter` component?

File: packages/frontend/src/app/submissions/page.tsx [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+  useEffect(() => {
+    try {
+      betterFetch('GET', 'http://localhost:3000/api/submissions').then(
+        (submissions) => {
+          setSubmissions(submissions)
+        },
+      )
+    } catch (error) {
+      setSubmissions([])
+    }
```
Comment (jakubkoci) [inline packages/frontend/src/app/submissions/page.tsx:41]: The `betterFetch` function is not awaited, so I'm not sure about the `catch` here. If I understand it correctly the error from `betterFetch` is not handled at all.

File: packages/frontend/src/app/submissions/page.tsx [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+          <th className="p-4 w-5/12">Company name</th>
+          <th className="p-4 w-1/12 text-right">State</th>
+          <th className="p-4 w-2/12 text-right">Updated</th>
+          <th className="p-4 w-2/12 text-right">Submitted</th>
+          <th className="p-4 w-2/12 text-right">Email</th>
+        </TableHeader>
+        <TableBody>
+          {submissions
+            .filter((item: Submission) => {
+              return !filterValues.length || filterValues.includes(item.state)
```
Comment (jakubkoci) [inline packages/frontend/src/app/submissions/page.tsx:79]: Why do we check `!filterValues.length`? 🤔

Comment (andrii-uhryn-absa) [reply packages/frontend/src/app/submissions/page.tsx:79]: When `filterValues` array is empty we should show all `submissions`.

Comment (andrii-uhryn-absa) [reply packages/frontend/src/app/submissions/page.tsx:79]: I discussed this logic with Marek.

File: packages/frontend/src/app/submissions/page.tsx (hunk 6 above)
Comment (andrii-uhryn-absa) [reply packages/frontend/src/app/submissions/page.tsx:41]: You are right. I'll fit it.

File: packages/frontend/src/app/page.tsx (hunk 1 above)
Comment (andrii-uhryn-absa) [reply packages/frontend/src/app/page.tsx]: Removed after refactoring

File: packages/frontend/src/app/submissions/page.tsx [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+  }
+
+  function isActive(filter: string): boolean {
+    return selectedOptions.includes(filter)
+  }
+
+  return (
+    <div className="flex flex-col">
+      <div className="pl-4 h-6 mb-2">
+        <TextSmBold className="h-6">Filters</TextSmBold>
```
Comment (jakubkoci) [inline packages/frontend/src/app/submissions/page.tsx:139]: Could we find out a way that we can style it with parent and not set `h-6` to the text element?

**Review by jakubkoci — APPROVED** [review]
Nice 👍

**Pull request merged**